client/src/api/users.js

```diff
@@ -21,3 +9,15 @@ export async function postUser(name) {
 
   return response;
 }
+
+export async function getUserById(id) {
+  const response = await fetch(`api/users/${id}`);
+  const user = await response.json();
+  return user;
+}
+
+export async function getUsers() {
+  const response = await fetch("api/users");
```

Comment:
This will be an issue if you call the api from a subroute like `https://app.com/some-domain` because it will result in  `https://app.com/some-domain/api/users` instead of  `https://app.com/api/users`
```suggestion
  const response = await fetch("/api/users");
```

This happens in other functions too.